File: COPA-MUNDIAL-FRONT/src/app/modulo04/models/equipo.ts
import { Pais } from './pais';
import { I18nEquipo } from './i18n_equipo';

export class Equipo {
    private _pais: Pais;
    private _descripcion: I18nEquipo;
    private _status: boolean;    // el equipo participa aun o no en la competencia
    private _grupo: string;
    private _habilitado: boolean; // atributo para funcionalidad de que no hayan jugadores sin equipo

    // getters y setters de la clase
    get getPais(): Pais {
        return this._pais;
    }

    get getDescripcion(): I18nEquipo {
        return this._descripcion;
    }

    get getStatus(): boolean {
        return this._status;
    }

    get getGrupo(): string {
        return this._grupo;
    }

    get getHabilitado(): boolean {
        return this._habilitado;
    }

    set setPais(pais: Pais) {
        this._pais = pais;
    }

    set setDescripcion(descripcion: I18nEquipo) {
        this._descripcion = descripcion;
    }

    set setStatus(status: boolean) {
        this._status = status;
    }

    set setGrupo(grupo: string) {
        this._grupo = grupo;
    }

    set setHabilitado(habilitado: boolean) {
        this._habilitado = habilitado;
    }
}

File: WebAPI/WebAPI/Models/Equipo.cs
using System;
using System.Collections.Generic;

namespace WebAPI.Models
{
    /// <summary>
    /// Clase que contiene los atributos del equipo
    /// </summary>
    public class Equipo
    {
        private Pais _pais;
        private List<I18nEquipo> _descripcion ;
        private bool _status;
        private string _grupo;
        private bool _habilitado;

        /// <summary>
        /// Constructor de la clase Equipo con parametros
        /// </summary>
        public Equipo(Pais pais, List<I18nEquipo> descripcion, bool status, string grupo, bool habilitado)
        {
            _pais = pais;
            _descripcion = descripcion;
            _status = status;
            _grupo = grupo;
            _habilitado = habilitado;
        }

        /// <summary>
        /// Constructor de la clase Equipo vacio
        /// </summary>
        public Equipo()
        {

        }
        
        /// <summary>
        /// Getters y Setters del atributo pais
        /// </summary>
        public Pais Pais
        {
            get { return _pais; }
            set { _pais = value; }
        }

        /// <summary>
        /// Getters y Setters del atributo descripcion
        /// </summary>
        public List<I18nEquipo> Descripcion
        {
            get { return _descripcion; }
            set { _descripcion = value; }
        }

        /// <summary>
        /// Getters y Setters del atributo status
        /// </summary>
        public bool Status
        {
            get { return _status; }
            set { _status = value; }
        }

        /// <summary>
        /// Getters y Setters del atributo grupo
        /// </summary>
        public string Grupo
        {
            get { return _grupo; }
            set { _grupo = value; }
        }

        /// <summary>
        /// Getters y Setters del atributo habilitado
        /// </summary>
        public bool Habilitado
        {
            get { return _habilitado; }
            set { _habilitado = value; }
        }
    }

}